src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -51,7 +51,19 @@ public CollectionModelBinder(IModelBinder elementBinder)
                 throw new ArgumentNullException(nameof(bindingContext));
             }
 
+            // For properties like the following, where the properties are get only but the actual model
+            // instance is an array, do not model bind.
+            //      public IEnumerable<> Addresses { get; } = new Address[] { };
+            //      public ICollection<> Addresses { get; } = new Address[] { };
```

Comment:
Offline decision was to leave current `IEnumerable<T> ... { get; } = new T[] {};` behaviour alone and let `ICollection<T> ... { get; } = new T[] {};` behaviour match.